packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/upgrade/upgrade.controller.js

```diff
@@ -0,0 +1,60 @@
+import find from 'lodash/find';
+import pick from 'lodash/pick';
+
+import { MODES } from '../components/termsOfUse/terms-of-use.constants';
+import { BACKUP_NAME } from './upgrade.constants';
+
+export default class {
+  /* @ngInject */
+  constructor($translate, dedicatedCloudDatacenterBackupService) {
+    this.$translate = $translate;
+    this.dedicatedCloudDatacenterBackupService = dedicatedCloudDatacenterBackupService;
+    this.MODES = MODES;
+    this.BACKUP_NAME = BACKUP_NAME;
+  }
+
+  $onInit() {
+    this.upgrading = false;
+    if (this.backup && this.backup.backupOffer) {
+      this.selectedOffer = find(this.backupOffers, {
+        offerName: this.backup.backupOffer,
+      });
+    }
+    this.conditionsAccepted = false;
+  }
+
+  upgrade() {
+    this.upgrading = true;
+    return this.dedicatedCloudDatacenterBackupService
+      .updateBackupCapabilities(
+        this.productId,
+        this.datacenterId,
+        pick(this.backup, [
+          'backupDurationInReport',
+          'backupSizeInReport',
+          'diskSizeInReport',
+          'fullDayInReport',
+          'restorePointInReport',
+          'mailAddress',
+          'backupOffer',
+        ]),
+      )
+      .then(() =>
+        this.goToBackup(
+          [
+            this.$translate.instant(
+              'dedicatedCloud_datacenter_backup_upgrade_success_1',
+              { offer: this.backup.backupOffer },
+            ),
+            this.$translate.instant(
+              'dedicatedCloud_datacenter_backup_upgrade_success_2',
+            ),
+          ].join(' '),
+        ),
+      )
+      .catch((error) => this.handleOrderError(error))
```

Comment:
where is handleOrderError defined?

Reply:
Functional doc describes what to do in case upgrade/downgrade fails. Can you please check it.

Reply:
There is no mention of error cases in specs and API also does not return error messages. The API is alway successful as it creates the task and the task might succeed or fail